src/test/kotlin/it/pagopa/ecommerce/scheduler/client/PaymentGatewayClientTest.kt

```diff
@@ -0,0 +1,94 @@
+package it.pagopa.ecommerce.scheduler.client
+
+import it.pagopa.ecommerce.scheduler.exceptions.BadGatewayException
+import it.pagopa.ecommerce.scheduler.exceptions.TransactionNotFound
+import it.pagopa.ecommerce.scheduler.utils.TestUtil.Companion.getMockedRefundRequest
+import it.pagopa.generated.ecommerce.gateway.v1.api.PaymentTransactionsControllerApi
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpHeaders
+import org.springframework.test.context.TestPropertySource
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+import java.nio.charset.Charset
+import java.util.*
+
+@SpringBootTest
+@TestPropertySource(locations = ["classpath:application.test.properties"])
+class PaymentGatewayClientTest {
+
+    @Mock
+    private lateinit var paymentTransactionsControllerApi: PaymentTransactionsControllerApi
+
+    @InjectMocks
+    private lateinit var paymentGatewayClient: PaymentGatewayClient
+
+    @Test
+    fun pgsClient_requestRefund_200(){
+        val testUIID: UUID = UUID.randomUUID()
+
+        // preconditions
+        Mockito.`when`(paymentTransactionsControllerApi.refundRequest(testUIID))
+            .thenReturn(Mono.just(getMockedRefundRequest(testUIID.toString())))
+
+        // test
+        val response = paymentGatewayClient.requestRefund(testUIID).block()
+
+        // asserts
+        assertEquals("success", response?.refundOutcome)
+    }
+
+    @Test
+    fun pgsClient_requestRefund_404(){
+        val testUIID: UUID = UUID.randomUUID()
+
+        // preconditions
+        Mockito.`when`(paymentTransactionsControllerApi.refundRequest(testUIID))
+            .thenReturn(Mono.error(WebClientResponseException.BadRequest.create(
+                404,"", HttpHeaders.EMPTY, ByteArray(0), Charset.defaultCharset())))
+
+        // test
+        val exc = assertThrows(Exception::class.java) {
+            paymentGatewayClient.requestRefund(testUIID).block()
+        }
+
+        assertInstanceOf(TransactionNotFound::class.java, exc)
```

Comment:
Is it different from `assertThrows<TransactionNotFound> { /* ... */ }`?

Reply:
It should be equivalent